contributing.md

```diff
@@ -43,8 +43,19 @@ here](https://join.slack.com/t/bazelbuild/shared_invite/enqtndyyodaxndy4mjqxlwu1
    takes too long (e.g. reviewer is unresponsive), please also send an email to
    [bazel-dev@googlegroups.com](mailto:bazel-dev@googlegroups.com).
 1. A Bazel maintainer at Google will apply the patch to our internal version
-   control system. The patch is exported as a Git commit, at which point the
-   GitHub pull request is closed.
+   control system.
+
+   This triggers internal presubmit checks like
+   [code linters](https://en.wikipedia.org/wiki/Lint_(software)), which may
+   suggest more changes. If you haven't expressed a preference, the submitter
+   will add "trivial" changes that don't affect design, which will also be
+   attributed to you. If you express preferences, such as "*no changes, not even
+   trivial ones*", these will be honored and any further changes will be applied
+   in followup PRs.
```

Comment:
I find it pretty hard to agree with this last part. I'm not comfortable with merging someone else's code knowing that there are obvious things to fix in it and that they'll require future attention. This is mostly a failure of our tooling: things would be different if we could use a chain of Git commits and merge those at once because then we could apply both the original submission and the cleanup atomically, with proper attribution for each. Unfortunately, we cannot do this.

If the PR author doesn't want the reviewer to touch the code at all, then I think the PR author should agree to perform all fixes *upfront* (as has traditionally been the case, I think?). Could we state that instead?

Reply:
Thinking over your other comment, let me argue against myself. :)

What do we do with the scenario where a PR author doesn't want changes and internal presubmits fail on failed *integration tests*? That's a bad scenario to be in. Semantic changes are really important in that case, and here it does seem worthwhile to push back on the original PR.

Reply:
I am not comfortable with "and any further changes will be applied in followup PRs.".
If it won't pass lint or integration tests, we should bounce the PR back with an explanation of the error and wait for the author to correct it. If an author is going to be extremely concerned about the presentation of a PR (as opposed to functionality of it), it is not fair for them to put the burden of making it comply with coding standards on the person doing the merge (who would have to write the followup).

Reply:
Sorry. I was not clear. I meant that only in the context of the author saying "*no changes, not even trivial ones*". I don't think an author should be able to say. "don't make trivial changes - I will fix them in a later change".   That allows someone to submit things that need fixing and then not bother with the fix.

Reply:
Updated. PTAL.